typescript/core/src/McapWriter.ts

```diff
@@ -34,6 +37,7 @@ export type McapWriterOptions = {
   startChannelId?: number;
   chunkSize?: number;
   compressChunk?: (chunkData: Uint8Array) => { compression: string; compressedData: Uint8Array };
+  appendMode?: boolean;
```

Comment:
If this isn't meant to be set to true manually, ideally it would not be exposed in the TS types. I guess TS doesn't allow you to define multiple constructors where some are private – so I wonder if it would make sense to remove this from the constructor entirely and set #appendMode later in the InitializeInAppendMode method?